integration_test/dr-buckets.tf.template

```diff
@@ -0,0 +1,154 @@
+resource "aws_s3_bucket" "orca-archive-worm" {
+  bucket = "PREFIX-orca-archive-worm"
+  force_destroy = true
+  tags = {
+    Deployment = "PREFIX"
+    Application = "ORCA"
+  }
+}
+resource "aws_s3_bucket_versioning" "orca-archive-worm-versioning" {
+  bucket = aws_s3_bucket.orca-archive-worm.id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+resource "aws_s3_bucket" "orca-reports" {
+  bucket = "PREFIX-orca-reports"
+  depends_on = [aws_s3_bucket.orca-archive-worm]
+  force_destroy = true
+  tags = {
+    Deployment = "PREFIX"
+    Application = "ORCA"
+  }
+}
+resource "aws_s3_bucket_lifecycle_configuration" "orca-reports-lifecycle-rule" {
+  bucket = aws_s3_bucket.orca-reports.id
+  rule {
+    id = "delete objects after 30 days"
+    status = "Enabled"
+    expiration {
+      days = 30
+    }
+  }
+}
+data "aws_iam_policy_document" "orca-reports-policy-json" {
+  statement {
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::236859827343:root"]
+    }
+    actions = [
+        "s3:GetObject*",
+        "s3:GetBucket*",
+        "s3:ListBucket",
+        "s3:PutObject",
+        "s3:PutObjectAcl",
+        "s3:PutBucketNotification"
+    ]
+    resources = [
+        "arn:aws:s3:::PREFIX-orca-reports",
+        "arn:aws:s3:::PREFIX-orca-reports/*"
+        ]
+  } 
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["s3.amazonaws.com"]
+      }
+      actions = [
+                  "s3:PutObject"
+      ]
+      resources = [
+                  "arn:aws:s3:::PREFIX-orca-reports",
+                  "arn:aws:s3:::PREFIX-orca-reports/*"
+                  ]
+      condition {
+        test     = "StringEquals"
+        variable = "s3:x-amz-acl"
+        values = [
+          "bucket-owner-full-control"
+        ]
+      }
+      condition {
+        test     = "StringEquals"
+        variable = "aws:SourceAccount"
+        values   = ["782417781503"]
+      }
+      condition {
+        test = "ArnLike"
+        variable = "aws:SourceArn"
+        values = [
+          "arn:aws:s3:::PREFIX-*"
+        ]
+      }
+    }
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["s3.amazonaws.com"]
+      }
+      actions = [
+                  "s3:PutObject"
+      ]
+      resources = [
+                  "arn:aws:s3:::PREFIX-orca-reports",
+                  "arn:aws:s3:::PREFIX-orca-reports/*"
+                  ]
+      condition {
+        test     = "StringEquals"
+        variable = "s3:x-amz-acl"
```

Comment:
You can add in your storage-class condition where appropriate.

Reply:
shouldn't that policy be attached to only the glacier archive bucket that we currently don't have in terraform?

Reply:
added to orca-primary